test/formal/checked_uint_mul_12.py

```diff
@@ -25,13 +24,11 @@
 	# cast to full n_bits values
 	X = BVUnsignedUpCast(X_short, n_bits)
 	Y = BVUnsignedUpCast(Y_short, n_bits)
+	product_raw = MUL(X, Y)
+	#remove any overflown bits
+	product = BVUnsignedCleanupFunction(product_raw, type_bits)
 
-	# Constants
-	maxValue = BVUnsignedMax(type_bits, n_bits)
-
-	# Overflow check in YulUtilFunction::overflowCheckedIntMulFunction
-	overflow_check = AND(ISZERO(ISZERO(X)), GT(Y, DIV(maxValue, X)))
+	# Overflow check in YulUtilFunction::overflowCheckedIntMulFunctions
+	overflow_check = AND(ISZERO(ISZERO(X)), ISZERO(EQ(Y, DIV(product, X))))
```

Comment:
The actual check in ``YulUtilFunction::overflowCheckedIntMulFunction`` has this, doesn't it:

```suggestion
	overflow_check = ISZERO(OR(ISZERO(X), EQ(Y, DIV(product, X))))
```

Reply:
Yes, I am always forgetting to change the formal tests after changing `YulUtilFunctions.cpp`.
Done already.